scipy/sparse/tests/test_base.py

```diff
@@ -3635,6 +3635,20 @@ def test_binop_explicit_zeros(self):
         assert (a + b).nnz == 2
         assert a.multiply(b).nnz == 1
 
+    def test_explicit_zeros(self):
+        # Ensures that binary ops of addition and subtraction does not 
+        # remove the explicit zeros.
+        row = np.array([0, 3, 1, 0])
+        col = np.array([0, 3, 1, 2])
+        data1 = np.array([0, 5, 7, 9])
+        data2 = np.array([0, 4, 6, 8])
+        m1 = coo_matrix((data1, (row, col)), shape=(4, 4))
+        m2 = coo_matrix((data2, (row, col)), shape=(4, 4))
```

Comment:
This is the CSR matrix test class, so these should be CSR matrices.

Reply:
You are right, I will change them to CSR matrices.